src/main/scala/com/databricks/sparkdl/ImageUtils.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2017 Databricks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.databricks.sparkdl
+
+import java.awt.image.BufferedImage
+import java.awt.{Color, Image}
+
+import org.apache.spark.sql.Row
+
+object ImageUtils {
+
+  /**
+   * Resizes and image and returns it as an Array[Byte]. Only 1 and 3 channel inputs, where each
+   * channel is a single Byte, are currently supported.
+   *
+   * @param srcHeight height of input image
+   * @param srcWidth width of input image
+   * @param srcChannels number of channels for input image (1 or 3)
+   * @param tgtHeight desired height of output image
+   * @param tgtWidth desired width of output image
+   * @param tgtChannels number of channels of output image (must be 3), may be used later to
+   *                    support more channels.
+   * @param imageData buffer containing image data.
+   * @return Array[Byte] of image data, if the input was BGR or 1 channel, the output will be BGR.
+   */
+  private[sparkdl] def resizeImage(
+      srcHeight: Int,
+      srcWidth: Int,
+      srcChannels: Int,
+      tgtHeight: Int,
+      tgtWidth: Int,
+      tgtChannels: Int,
+      imageData: Array[Byte]) = {
+    require(tgtChannels == 3, "output must have 3 channels.")
+    val srcImg = new BufferedImage(srcHeight, srcWidth, BufferedImage.TYPE_3BYTE_BGR)
+
+    var offset, h = 0
+    while (h < srcHeight) {
+      var w = 0
+      while (w < srcWidth) {
+        val (r, g, b): (Byte, Byte, Byte) = srcChannels match {
```

Comment:
Let's refactor this code into two loops, one for `srcChannels == 1` and one for `srcChannels == 3`. 
This way we only have to deal with the condition once outside the loop, making the code more performant.